windows-gaming-driver/src/controller.rs

```diff
@@ -90,18 +100,42 @@ impl Controller {
         // we return false if we didn't notice the GA going down as the timer still
         // exists in that case so it would be a bug to create a second one.
 
-        self.ga_pong_expected = false;
-
-        if self.ga_up {
-            return false;
+        let ga = self.ga;
+        self.ga = State::Up;
+        match ga {
+            State::Pinging | State::Up => false,
+            State::Down => true,
+            State::Resuming => {
+                self.set_io_attached(true, false);
+                true
+            }
+            State::Suspending => false, // wtf though
         }
+    }
 
-        self.ga_up = true;
-        true
+    pub fn ga_suspending(&mut self) {
+        self.set_io_attached(false, true);
+        self.ga = State::Suspending;
     }
 
     pub fn set_io_attached(&mut self, state: bool, force: bool) {
```

Comment:
[C-CUSTOM-TYPE](https://github.com/brson/rust-api-guidelines/blob/master/README.md#arguments-convey-meaning-through-types-not-bool-or-option-c-custom-type)